### PR TITLE
[DOC][TS SDK] add documentation for Provider and IndexerClient classes

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
This PR adds documentation for Provider and IndexerClient classes on the TS SDK

### CODE REVIEW

**Comment by clay-aptos** [comment]
Thanks so much, @0xmaayan ! I am asking @christian-aptos to conduct initial review. We will of course want to announce these changes in See what's new. Christian can help with that. Thank you both!

File: developer-docs-site/docs/sdks/ts-sdk/typescript-sdk-overview.md [hunk 1]
```diff
@@ -89,18 +90,52 @@ The transaction builder contains the Typescript types for constructing the trans
 1. EntryFunction
 2. Script
 
+### Provider Class
+
+To help developers and to provide a better dev experience, the SDK provides a [Provider class](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/providers/provider.ts), that extends both `AptosClient` and `IndexerClient` classes and gives the end user the option to simply create a `Provider` instance and call a method by hiding the underline implementation.
+
+Provider class accepts
+
+- `network` - Netowrk enum type - `mainnet | testnet | devnet`.
```
Comment (clay-aptos) [inline developer-docs-site/docs/sdks/ts-sdk/typescript-sdk-overview.md:99]: misspelled as "Netowrk"

File: developer-docs-site/docs/sdks/ts-sdk/typescript-sdk-overview.md [hunk 2]
```diff
@@ -89,18 +90,52 @@ The transaction builder contains the Typescript types for constructing the trans
 1. EntryFunction
 2. Script
 
+### Provider Class
+
+To help developers and to provide a better dev experience, the SDK provides a [Provider class](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/providers/provider.ts), that extends both `AptosClient` and `IndexerClient` classes and gives the end user the option to simply create a `Provider` instance and call a method by hiding the underline implementation.
```
Comment (clay-aptos) [inline developer-docs-site/docs/sdks/ts-sdk/typescript-sdk-overview.md:95]: underline should be underlying

**Review by saharct — APPROVED** [review]
Hi @0xmaayan , it looks good to me with some minor changes!

**Pull request merged**